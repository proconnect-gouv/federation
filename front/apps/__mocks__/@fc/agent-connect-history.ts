export const useUserHistory = jest.fn(() => [
  { active: false, display: true, name: 'mock-name-1', uid: 'mokc-uid-1' },
  { active: false, display: true, name: 'mock-name-2', uid: 'mokc-uid-2' },
  { active: false, display: true, name: 'mock-name-3', uid: 'mokc-uid-3' },
  { active: false, display: true, name: 'mock-name-4', uid: 'mokc-uid-4' },
]);

export const useAddToUserHistory = jest.fn(() => jest.fn());

export const useRemoveFromUserHistory = jest.fn(() => jest.fn());
