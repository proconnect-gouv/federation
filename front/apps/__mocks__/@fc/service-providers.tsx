export const ServiceProviderItemComponent = jest.fn(({ datapassId }) => (
  <div>ServiceProviderItemComponent {datapassId}</div>
));

export {
  ServiceProviderEditActionTypes,
  ServiceProvidersActionTypes,
  ServiceProvidersPermissionTypes,
  ServiceProviderViewActionTypes,
} from '@fc/service-providers';
