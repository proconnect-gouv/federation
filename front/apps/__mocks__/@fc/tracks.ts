export const TracksListComponent = jest.fn();
