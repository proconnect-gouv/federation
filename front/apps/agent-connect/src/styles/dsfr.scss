// @TODO a supprimer une fois la lib dsfr partag√©e entre les apps front

.mr8 {
  margin-right: 8px;
}

.is-block {
  display: block;
}

.flex-columns {
  display: flex;
  flex-direction: row;
}

.items-center {
  align-items: center;
}

a.no-underline {
  text-decoration: none;
}

.bg-blue-france-100 {
  background-color: #f2f2f9;
}

.is-blue-agentconnect {
  color: #0579ee;
}

.fr-display-xs {
  font-size: 2.5rem;
  line-height: 3rem;
}

h1 {
  font-size: 2rem;
  line-height: 2.5rem;
}

.content-wrapper-sm {
  margin-left: auto;
  margin-right: auto;
}

.my32 {
  margin-bottom: 2rem;
  margin-top: 2rem;
}

.mb32 {
  margin-bottom: 2rem;
}

.px120 {
  padding-left: 7.5rem;
  padding-right: 7.5rem;
}

.fr-text-lg {
  font-size: 1.125rem;
  line-height: 1.75rem;
}

.p24 {
  padding: 1.5rem;
}

.no-no-padding {
  padding: 0 !important;
}

@media (min-width: 768px) {
  .fr-display-xs {
    font-size: 3rem;
    line-height: 3.5rem;
  }

  h1 {
    font-size: 2.5rem;
    line-height: 3rem;
  }

  .content-wrapper-sm {
    max-width: 792px;
  }

  .fr-text-lg {
    font-size: 1.25rem;
    line-height: 2rem;
  }
}
