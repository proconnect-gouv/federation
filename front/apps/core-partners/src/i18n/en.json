{
  "CorePartners.instance.createdAt": "Created at : {date}"
}
