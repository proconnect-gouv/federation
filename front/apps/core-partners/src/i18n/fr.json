{
  "CorePartners.instance.createdAt": "Créée le : {date}",
  "Form.FORM_ERROR": "Veuillez réessayer ultérieurement"
}
