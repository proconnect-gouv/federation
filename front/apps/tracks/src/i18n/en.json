{
  "TracksPage.previousPage": "Previous page",
  "TracksPage.nextPage": "Next page"
}
