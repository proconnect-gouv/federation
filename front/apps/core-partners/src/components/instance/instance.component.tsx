import { DateTime } from 'luxon';
import React from 'react';

import type { ISODate, PropsWithClassName } from '@fc/common';
import { CardComponent, Sizes } from '@fc/dsfr';
import { t } from '@fc/i18n';

import type { VersionInterface } from '../../interfaces';

interface InstanceComponentProps extends PropsWithClassName {
  id: string;
  createdAt: ISODate;
  data: VersionInterface['data'];
}

export const InstanceComponent = React.memo(
  ({ className, createdAt, data, id }: InstanceComponentProps) => {
    const formattedDate = DateTime.fromISO(createdAt, { zone: 'utc' }).toFormat('dd/MM/yyyy');
    const date = t('CorePartners.instance.createdAt', { date: formattedDate });

    const { client_id: clientId, client_secret: clientSecret, name } = data;

    return (
      <CardComponent
        enlargeLink
        className={className}
        details={{
          top: {
            className: 'fr-icon-arrow-right-line',
            content: date,
          },
        }}
        link={id}
        size={Sizes.LARGE}
        title={name}>
        <p>
          <b className="is-block">Client ID</b>
          <span data-testid="InstanceComponent-client-id">{clientId}</span>
        </p>
        <p>
          <b className="is-block">Client Secret</b>
          <span data-testid="InstanceComponent-client-secret">{clientSecret}</span>
        </p>
      </CardComponent>
    );
  },
);

InstanceComponent.displayName = 'InstanceComponent';
