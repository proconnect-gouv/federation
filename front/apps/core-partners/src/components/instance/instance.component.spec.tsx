import { render } from '@testing-library/react';

import type { ISODate } from '@fc/common';
import { CardComponent } from '@fc/dsfr';
import { t } from '@fc/i18n';

import type { Environment } from '../../enums';
import { InstanceComponent } from './instance.component';

describe('InstanceComponent', () => {
  // Given
  const instanceItemMock = {
    createdAt: 'any-date-mock' as unknown as ISODate,
    environment: 'Sanbox' as unknown as Environment,
    id: 'any-id-mock',
    name: 'any-name-mock',
    updatedAt: 'any-date-mock' as unknown as ISODate,
    versions: [expect.any(Object)],
  };

  beforeEach(() => {
    // Given
    jest.mocked(t).mockReturnValue('any-date-value-mock');
  });

  it('should match the snapshot', () => {
    // When
    const { container } = render(<InstanceComponent item={instanceItemMock} />);

    // Then
    expect(t).toHaveBeenCalledOnce();
    expect(t).toHaveBeenCalledWith('CorePartners.instance.createdAt', { date: 'any-date-mock' });
    expect(container).toMatchSnapshot();
    expect(CardComponent).toHaveBeenCalledOnce();
    expect(CardComponent).toHaveBeenCalledWith(
      {
        details: {
          top: {
            className: 'fr-icon-arrow-right-line',
            content: 'any-date-value-mock',
          },
        },
        enlargeLink: true,
        link: 'any-id-mock',
        size: 'lg',
        title: 'any-name-mock',
      },
      {},
    );
  });
});
