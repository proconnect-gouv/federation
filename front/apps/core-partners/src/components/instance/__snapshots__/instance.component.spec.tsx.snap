// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstanceComponent should match the snapshot 1`] = `
<div>
  <div>
    CardComponent
  </div>
</div>
`;
